### PR TITLE
Add getOverlays(CollectionGroup)

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Splitting up #3323. This is the second PR.

### CODE REVIEW

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/DocumentOverlayCache.java [hunk 1]
```diff
@@ -56,4 +56,16 @@
...
+   * Returns {@code count} overlays with a batch ID higher than {@code sinceBatchId} for the
+   * provided collection group, processed by ascending batch ID. The method always returns all
+   * overlays for a batch even if a batch contains more documents than the remaining limit.
+   *
+   * @param collectionGroup The collection group to get the overlays for.
+   * @param sinceBatchId The minimum batch ID to filter by (exclusive). Only overlays that contain a
+   *     change past `sinceBatchId` are returned.
+   * @return Mapping of each document key in the collection group to its overlay.
+   */
+  Map<DocumentKey, Overlay> getOverlays(String collectionGroup, int sinceBatchId, int count);
```
Comment (thebrianchen) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/local/DocumentOverlayCache.java:70]: optional: include `count` as `@param`

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteDocumentOverlayCache.java [hunk 2]
```diff
@@ -99,6 +99,51 @@ public void removeOverlaysForBatchId(int batchId) {
     return result;
   }
 
+  @Override
```
Comment (thebrianchen) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteDocumentOverlayCache.java:102]: Love this implementation. So much cleaner!

**Review by thebrianchen — APPROVED** [review]
lgtm!

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/DocumentOverlayCache.java (hunk 1 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/DocumentOverlayCache.java:72]: Done

**Pull request merged**